packages/page-treasury/src/Tips/SlashTipModal.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017-2021 @polkadot/app-treasury authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import React, { useCallback, useState } from 'react';
+
+import { InputAddress, InputNumber, Modal, TxButton } from '@polkadot/react-components';
+import { useApi, useCollectiveInstance, useCollectiveMembers } from '@polkadot/react-hooks';
+import { BN_ZERO } from '@polkadot/util';
+
+import { useTranslation } from '../translate';
+
+interface Props {
+  closeModal: () => void,
+  hash: string;
+}
+
+interface Threshold {
+  isThresholdValid: boolean;
+  threshold?: BN;
+}
+
+function SlashTipModal ({ closeModal, hash }: Props): React.ReactElement<Props> | null {
+  const [accountId, setAccountId] = useState<string | null>(null);
+  const [{ isThresholdValid, threshold }, setThreshold] = useState<Threshold>({ isThresholdValid: false });
+  const modLocation = useCollectiveInstance('council');
+  const { members } = useCollectiveMembers('council');
+  const { api } = useApi();
+  const proposal = (api.tx.tips || api.tx.treasury).slashTip(hash);
```

Comment:
Rather use a memo for this, otherwise it gets executed each render -

```js
const proposal = useMemo(() => (api.tx.tips || api.tx.treasury).slashTip(hash), [hash]);
```